src/main/java/com/faforever/moderatorclient/api/domain/MessagesService.java

```diff
@@ -33,26 +33,43 @@ public MessagesService(FafApiCommunicationService fafApi, MessagesMapper message
     }
 
     private List<Message> getAllMessagesFromApi() {
-        return fafApi.getAll(ElideRouteBuilder.of(Message.class));
+        return fafApi.getAll(ElideNavigator.of(Message.class).collection());
     }
 
     public void updateMessage(MessageFx messageFx) {
         Message message = messagesMapper.map(messageFx);
-        fafApi.patch(ElideRouteBuilder.of(Message.class).id(message.getId()), message);
+        fafApi.patch(ElideNavigator.of(Message.class).id(message.getId()), message);
         applicationEventPublisher.publishEvent(new MessagesChangedEvent());
     }
 
-    public void deleteCategory(MessageFx selectedItem) {
+    public void deleteMessage(MessageFx selectedItem) {
         Message message = messagesMapper.map(selectedItem);
-        fafApi.delete(ElideRouteBuilder.of(Message.class).id(message.getId()));
+        fafApi.delete(ElideNavigator.of(Message.class).id(message.getId()));
         applicationEventPublisher.publishEvent(new MessagesChangedEvent());
 
     }
 
     public Message createMessage(MessageFx messageFx) {
         Message message = messagesMapper.map(messageFx);
-        Message result = fafApi.post(ElideRouteBuilder.of(Message.class), message);
+        Message result = fafApi.post(ElideNavigator.of(Message.class).collection(), message);
         applicationEventPublisher.publishEvent(new MessagesChangedEvent());
         return result;
     }
+
+    public Message putMessage(MessageFx messageFx) {
+        Message message = messagesMapper.map(messageFx);
+        List<Message> messages = fafApi.getAll(ElideNavigator.of(Message.class).collection()
+                .addFilter(ElideNavigator.qBuilder().string("region").eq(message.getRegion()))
+                .addFilter(ElideNavigator.qBuilder().string("language").eq(message.getLanguage()))
+                .addFilter(ElideNavigator.qBuilder().string("key").eq(message.getKey())));
```

Comment:
My bad. This should not be `addFilter`, but `setFilter` instead in the ElideNavigator. You can't add multiple filters, each one will override the previous.
Use the qBuilders `.and()` concatenation instead.